spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/types/UserDefinedType.scala

```diff
@@ -32,6 +32,7 @@ case class UserDefinedType(name: String, columns: IndexedSeq[UDTFieldDef])
   def cqlTypeName = name
 
   val fieldConverters = columnTypes.map(_.converterToCassandra)
+  private val metadata = CassandraRowMetadata.fromColumnNames(columnNames)
```

Comment:
let's make it lazy

Reply:
@jtgrabowski I'll push the changes soon. Anyway, it seems that it is not possible to achieve the full binary compatibility with this change. I've recalled the 'copy' method of the case classes. its signature will also change. :(

Reply:
done